### ISSUE TITLE
Crashes with "Application fatal error" after today's TruckersMP update

### ISSUE THREAD

**Issue by TurboWafflz** [truckersmp-cli/truckersmp-cli#161]
## Describe the bug
After today's (3/3/2021) TruckersMP update, when starting the game it crashes and displays an error saying "Application Fatal Error". According to the TruckersMP log, the cause of the crash was EXCEPTION_ACCESS_VIOLATION.
[TruckersMP crash log](https://github.com/lhark/truckersmp-cli/files/6080045/last_crash.log)

## Expected behavior
The game to start normally as it did before the update

## Command
I tried both 
~~~
truckersmp-cli --ets2 --start -n TurboWafflz -i 1420170
~~~
and 
~~~
truckersmp-cli --ets2 --start -n TurboWafflz -i 1245040
~~~
## Terminal output
<details><summary>Collapsed log</summary>

~~~
** DEBUG **  No moddir set, setting to default
** INFO **  Mod directory: /home/fmw/.local/share/truckersmp-cli/TruckersMP
** INFO **  Platform is Linux, using Proton
** INFO **  AppID/GameID: 227300 (ets2)
** INFO **  Game directory: /home/fmw/.local/share/truckersmp-cli/Euro Truck Simulator 2/data
** INFO **  Prefix: /home/fmw/.local/share/truckersmp-cli/Euro Truck Simulator 2/prefix
** INFO **  Proton directory: /home/fmw/.local/share/truckersmp-cli/Proton
** DEBUG **  Updating mod files
** DEBUG **  Creating directory /home/fmw/.local/share/truckersmp-cli/TruckersMP if it doesn't exist
** DEBUG **  No files to download
** DEBUG **  Starting game with Proton
** DEBUG **  Steam is running
** INFO **  Steam installation directory: /home/fmw/.local/share/Steam
** DEBUG **  Creating directory /home/fmw/.local/share/truckersmp-cli/Euro Truck Simulator 2/prefix if it doesn't exist
** DEBUG **  winediscordipcbridge.exe is present, MD5 is OK.
** INFO **  Starting winediscordipcbridge.exe
** INFO **  Startup command:
  SteamGameId=227300
  SteamAppId=227300
  STEAM_COMPAT_DATA_PATH=/home/fmw/.local/share/truckersmp-cli/Euro Truck Simulator 2/prefix
  STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/fmw/.local/share/Steam
  PROTON_USE_WINED3D=0
  PROTON_NO_D3D11=1
  LD_PRELOAD=libgtk3-nocsd.so.0:/home/fmw/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so
  /usr/bin/python3 /home/fmw/.local/share/truckersmp-cli/Proton/proton
  run
  /home/fmw/.local/lib/python3.8/site-packages/truckersmp_cli/truckersmp-cli.exe /home/fmw/.local/share/truckersmp-cli/Euro Truck Simulator 2/data /home/fmw/.local/share/truckersmp-cli/TruckersMP
~~~
</details>

## Screenshots
<details><summary>Collapsed screenshots</summary>

![Screenshot_20210303_194548](https://user-images.githubusercontent.com/26664670/109893234-4cf52580-7c59-11eb-9dd7-dcf129f7910a.png)

</details>

## Information
 - OS: Ubuntu 20.04
 - Proton or Wine version: Tested both Proton 5.0 and 5.13 and got same behavior
 - truckersmp-cli version: 0.5.0

**Comment by TabulateJarl8** [comment]
This same issue happens on Manjaro, same log error as well. I've also tested running ATS, and that has the same starting issue that ETS2 is having here.

- Kernel: Linux 5.10.18-1-MANJARO
- Proton or Wine version: Tested both Proton 5.0 and 5.13, doesn't work with either
- truckersmp-cli version: 0.5.0

**Comment by TurboWafflz** [comment]
Does anyone know if this updated added the new "Alternating Vehicles" feature? I was thinking that that could possibly be the source of the problem if it requires additional files which truckersmp-cli is not currently setup to download.

**Comment by Johnahon010** [comment]
Same issue as well

- OS: Ubuntu 18.04
- Proton or Wine version: Tested both Proton 5.0 and 5.13 and got same behavior
- truckersmp-cli version: 0.4.1

**Comment by section1** [comment]
hello
i have same problem in gentoo... some hours before was working perfectly.
i'm using version 0.5.0 of truckersmp-cli-0.5.0
if need system info, versions, etc. i can provide it.
similar crash log as TurboWafflz.
[crash.log](https://github.com/lhark/truckersmp-cli/files/6082184/crash.log)

**Comment by JoohanV** [comment]
The changelog for the update reads as follows:

Client Changelog 0.2.3.2.1
- Hid police and Patreon accessories in the truck/trailer configuration if the user does not have permission to equip them.
- Added a chat message about an incorrect/corrupted modification package on joining (when being kicked for this reason).
- Fixed the truck's liftable axle constant sound noises.
- Other internal fixes and improvements.

Client Changelog 0.2.3.2.1 - Hotfix N.1
- If the server has the option "police cars for players" enabled, anybody can see the police accessories.

I am thinking that particularly the first one of these changes might cause problems if the files in question are hardcoded in truckersmp-cli to be used when starting the game. It might also be something in the "other internal fixes and improvements" though, so it might be useful to try and get a list of what those contain exactly...

**Comment by JoohanV** [comment]
I also just tested if I was still able to play, and I was still able to get into the game and join a server.

**Comment by kakurasan** [comment]
It crashes for me only when OpenGL is used. Sometimes TruckersMP OpenGL backend stops working when updating and it starts working again later.

Nobody knows whether the issue is temporary. If not, we would need to change the default backend to D3D11.

**Comment by TurboWafflz** [comment]
> It crashes for me only when OpenGL is used. Sometimes TruckersMP OpenGL backend stops working when updating and it starts working again later.

It still doesn't work for me with DirectX, the game starts but the TruckersMP overlay doesn't appear and I can't move the cursor

**Comment by TabulateJarl8** [comment]
> > It crashes for me only when OpenGL is used. Sometimes TruckersMP OpenGL backend stops working when updating and it starts working again later.
> 
> It still doesn't work for me with DirectX, the game starts but the TruckersMP overlay doesn't appear and I can't move the cursor

This happens for me as well

**Comment by Lucki** [comment]
> It still doesn't work for me with DirectX, the game starts but the TruckersMP overlay doesn't appear and I can't move the cursor

> This happens for me as well

https://github.com/lhark/truckersmp-cli#direct3d-11-dxvk-or-wined3d

**Comment by TurboWafflz** [comment]
Ok, switching to `--activate-native-d3dcompiler-47` fixes the DirectX issue, sorry for confusing that with this issue, I hadn't tested DirectX before and just assumed it worked without any special treatment

**Comment by kallestrop** [comment]
when i start with 

`truckersmp-cli --ets2 --start --proton --enable-d3d11 --activate-native-d3dcompiler-47`

i get the same error. But i found out, that it will only happen when i have Mesa 21.1.0-devel (git-517600b4d5) active.
If i switch back to  Mesa 20.3.4 all works fine.

My system (Debian-Sid Bullseye):

Working config (Mesa 20.3.4):
Graphics:  Device-1: Advanced Micro Devices [AMD/ATI] Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] driver: amdgpu v: kernel 
           Display: x11 server: X.Org 1.20.10 driver: loaded: amdgpu,ati unloaded: fbdev,modesetting,radeon,vesa resolution: 
           1: 2560x1440~60Hz 2: 2560x1440~60Hz 3: 2560x1440~60Hz 
           OpenGL: renderer: AMD SIENNA_CICHLID (DRM 3.40.0 5.10.0-3-amd64 LLVM 11.0.1) v: 4.6 Mesa 20.3.4 


Not working config (Mesa 21.1.0-devel):
Graphics:  Device-1: Advanced Micro Devices [AMD/ATI] Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] driver: amdgpu v: kernel 
           Display: x11 server: X.Org 1.20.10 driver: loaded: amdgpu,ati unloaded: fbdev,modesetting,radeon,vesa resolution: 
           1: 2560x1440~60Hz 2: 2560x1440~60Hz 3: 2560x1440~60Hz 
           OpenGL: renderer: AMD SIENNA_CICHLID (DRM 3.40.0 5.10.0-3-amd64 LLVM 11.0.1) 
           v: 4.6 Mesa 21.1.0-devel (git-517600b4d5)

**Comment by kallestrop** [comment]
I found a solution for my problem ets2 not starting and abort with same error as described in the first post. 
I had to add NODEVICE_SELECT=1 as a start parameter. So after starting truckersmp-cli with

`NODEVICE_SELECT=1 truckersmp-cli --ets2 --start --proton --enable-d3d11 --activate-native-d3dcompiler-47`

game runs perfect again with newer Mesa 21.1.0-devel (tested with Proton 5.0)
Maybe all who have problems with this error may try to start with this option.
